packages/toolpad-app/src/utils/domView.ts

```diff
@@ -0,0 +1,45 @@
+import { NodeId } from '@mui/toolpad-core';
+import { matchPath } from 'react-router-dom';
+import { APP_PAGE_ROUTE, APP_CONNECTION_ROUTE, APP_CODE_COMPONENT_ROUTE } from '../routes';
+
+export type PageView =
+  | { kind: 'query'; nodeId: NodeId }
+  | { kind: 'pageModule' }
+  | { kind: 'pageParameters' };
+
+export type DomView =
+  | { kind: 'page'; nodeId?: NodeId; view?: PageView }
+  | { kind: 'connection'; nodeId: NodeId }
+  | { kind: 'codeComponent'; nodeId: NodeId };
+
+export function getPathnameFromView(appId: string, view: DomView): string | null {
+  switch (view.kind) {
+    case 'page':
+      return `/app/${appId}/pages/${view.nodeId}`;
+    case 'connection':
+      return `/app/${appId}/connections/${view.nodeId}`;
+    case 'codeComponent':
+      return `/app/${appId}/codeComponents/${view.nodeId}`;
+    default:
+      return null;
```

Comment:
Should there be any view for which no pathame exists? Perhaps this should throw?

Reply:
Sounds good! I've added it in a new commit to include small type adjustments since the result can't be `null`